packages/tpetra/core/test/FECrsMatrix/FECrsMatrix_UnitTests.cpp

```diff
@@ -292,13 +302,92 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( FECrsMatrix, Assemble1D, LO, GO, Scalar, Node
   TPETRA_GLOBAL_SUCCESS_CHECK(out,comm,success)
 }
 
+////
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( FECrsMatrix, Assemble1D_Kokkos, LO, GO, Scalar, Node )
+{
+  using exec_space = typename Node::execution_space;
+  using range_type = Kokkos::RangePolicy<exec_space, LO>;
+  using FEMAT = typename Tpetra::FECrsMatrix<Scalar,LO,GO,Node>;
+  using CMAT = typename Tpetra::CrsMatrix<Scalar,LO,GO,Node>;
+  using FEG = typename Tpetra::FECrsGraph<LO,GO,Node>;
+  using ImplScalarType = typename FEMAT::impl_scalar_type;
+
+  // get a comm
+  RCP<const Comm<int> > comm = getDefaultComm();
+  
+  // Generate a mesh
+  size_t numLocal = 10;
+  GraphPack<LO,GO,Node> pack;
+  generate_fem1d_graph(numLocal,comm,pack);
+
+  // Make the graph    
+  // FIXME: We should be able to get away with 3 for StaticProfile here, but we need 4 since duplicates are
+  // not being handled correctly. 
+  RCP<FEG> graph = rcp(new FEG(pack.uniqueMap,pack.overlapMap,4));
+
+  graph->beginFill();
+  for(size_t i=0; i<(size_t)pack.element2node.size(); i++) {
+    for(size_t j=0; j<pack.element2node[i].size(); j++) {
+      GO gid_j = pack.element2node[i][j];
+      for(size_t k=0; k<pack.element2node[i].size(); k++) {
+        GO gid_k = pack.element2node[i][k];
+        //printf("Inserting (%d,%d)\n",gid_j,gid_k);
+        graph->insertGlobalIndices(gid_j,1,&gid_k);
+      }
+    }
+  }
+  graph->endFill();
+
+  // Generate the "local stiffness matrix"
+  std::vector<std::vector<Scalar> > localValues = generate_fem1d_element_values<Scalar>();
+  auto kokkosValues = generate_fem1d_element_values_kokkos<Scalar, Node>();
+
+  // Make the matrix two ways
+  FEMAT mat1(graph); // Here we use graph as a FECrsGraph
+  CMAT mat2(graph);  // Here we use graph as a CrsGraph in OWNED mode
+  mat1.beginFill();
+
+  auto k_e2n = pack.k_element2node;
+  auto localMat = mat1.getLocalMatrix();
+  auto localMap = pack.overlapMap->getLocalMap();
+  //get local map too
+
+  Kokkos::parallel_for("assemble_1d",
+		       range_type (0,k_e2n.extent(0)), 
+		       KOKKOS_LAMBDA(const size_t i) {
```

Comment:
`range_type` uses `LO` (`int`) as the index type, but you're using `size_t` here.  This may cause a signed/unsigned warning.  It's also deceptive for readers, because Kokkos will hand your functor `int`, even though it claims to take `size_t`.